File: backend/src/main/java/com/arstem/backend/misconception/api/RecordMisconceptionRequest.java
package com.arstem.backend.misconception.api;

import com.arstem.backend.misconception.domain.MisconceptionSeverity;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record RecordMisconceptionRequest(
        @NotBlank(message = "Session id is required.")
        String sessionId,
        @NotBlank(message = "Topic code is required.")
        @Size(max = 100, message = "Topic code must be at most 100 characters.")
        String topicCode,
        @NotBlank(message = "Misconception code is required.")
        @Size(max = 100, message = "Misconception code must be at most 100 characters.")
        String misconceptionCode,
        @NotBlank(message = "Misconception title is required.")
        @Size(max = 200, message = "Misconception title must be at most 200 characters.")
        String misconceptionTitle,
        @NotBlank(message = "Description is required.")
        @Size(max = 2000, message = "Description must be at most 2000 characters.")
        String description,
        @NotNull(message = "Severity is required.")
        MisconceptionSeverity severity) {
}

File: frontend/lib/features/misconceptions/models/record_misconception_request.dart
/// Request model for recording a learner misconception.
class RecordMisconceptionRequest {
  final String sessionId;
  final String topicCode;
  final String misconceptionCode;
  final String misconceptionTitle;
  final String description;
  final String severity;

  RecordMisconceptionRequest({
    required this.sessionId,
    required this.topicCode,
    required this.misconceptionCode,
    required this.misconceptionTitle,
    required this.description,
    required this.severity,
  });

  Map<String, dynamic> toJson() {
    return {
      'sessionId': sessionId,
      'topicCode': topicCode,
      'misconceptionCode': misconceptionCode,
      'misconceptionTitle': misconceptionTitle,
      'description': description,
      'severity': severity,
    };
  }
}
